File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/semanticdata/optionvalues/KillModeOptionValue.java
package net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.optionvalues;

import com.google.common.collect.ImmutableSet;

import java.util.Set;

/**
 * This validator is for the config_kill_mode validator.
 * <p></p>
 * This seems to use the values referenced in:
 * <p></p>
 * kill.c ~ line 35.
 */
public class KillModeOptionValue implements OptionValueInformation {
  
  private static final Set<String> validOptions = ImmutableSet.of("control-group", "process", "mixed", "none");
  
  @Override
  public Set<String> getAutoCompleteOptions() {
    return validOptions;
  }
  
  @Override
  public String getErrorMessage(String value) {
    if (!validOptions.contains(value)) {
      return "The value supplied " + value + " does not match one of the expected values: " + validOptions;
    } else {
      return null;
    }
  }
  
  @Override
  public String getValidatorName() {
    return "config_parse_kill_mode";
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/semanticdata/optionvalues/KillModeOptionValue.kt
package net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.optionvalues

import com.google.common.collect.ImmutableSet
import com.intellij.openapi.project.Project
import net.sjrx.intellij.plugins.systemdunitfiles.semanticdata.Validator

/**
 * This validator is for the config_kill_mode validator.
 *
 *
 * This seems to use the values referenced in:
 *
 *
 * kill.c ~ line 35.
 */
class KillModeOptionValue : AbstractEnumOptionValue(validOptions, VALIDATOR_NAME) {

  companion object {
    private val validOptions: Set<String> = ImmutableSet.of("control-group", "process", "mixed", "none")
    const val VALIDATOR_NAME = "config_parse_kill_mode"
    val validators = mapOf(Validator(VALIDATOR_NAME, "0") to KillModeOptionValue())
  }
}


